airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -182,7 +174,59 @@ def test_oauth_flow_parameters(self, actual_connector_spec: ConnectorSpecificati
         diff = params - schema_path
         assert diff == set(), f"Specified oauth fields are missed from spec schema: {diff}"
 
-    def test_additional_properties_is_true(self, actual_connector_spec):
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_required_field_declaration(self, spec_diff):
+        added_required_fields = [
+            addition for addition in spec_diff.get("dictionary_item_added", []) if addition.path(output_format="list")[-1] == "required"
+        ]
+        assert len(added_required_fields) == 0, f"The current spec has a new required field: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_required_property(self, spec_diff):
+        added_required_properties = [
+            addition for addition in spec_diff.get("iterable_item_added", []) if addition.up.path(output_format="list")[-1] == "required"
+        ]
+        assert len(added_required_properties) == 0, f"The current spec has a new required property: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_type_field_changed_from_list_to_string(self, spec_diff):
+        type_changes = [
+            type_change for type_change in spec_diff.get("type_changes", []) if type_change.path(output_format="list")[-1] == "type"
+        ]
+        for type_change in type_changes:
+            if isinstance(type_change.t1, list) and isinstance(type_change.t2, str):
+                raise AssertionError(f"The current spec narrowed a field type: {spec_diff.pretty()}")
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_type_field_has_narrowed(self, spec_diff):
+        removals = [
+            removal for removal in spec_diff.get("iterable_item_removed", []) if removal.up.path(output_format="list")[-1] == "type"
+        ]
+        assert len(removals) == 0, f"The current spec narrowed a field type: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_enum_field_has_narrowed(self, spec_diff):
+        removals = [
+            removal for removal in spec_diff.get("iterable_item_removed", []) if removal.up.path(output_format="list")[-1] == "enum"
+        ]
+        assert len(removals) == 0, f"The current spec narrowed a field enum: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_enum_field_declaration(self, spec_diff):
+        added_enum_fields = [
+            addition for addition in spec_diff.get("dictionary_item_added", []) if addition.path(output_format="list")[-1] == "enum"
+        ]
+        assert len(added_enum_fields) == 0, f"The current spec has a new enum field: {spec_diff.pretty()}"
```

Comment:
This doesn't seem like a change that would always be breaking 🤔 

I can see changing an existing field from string to enum being breaking, but it should be fine to add an entirely new (optional) field that happens to be an enum. Could we add a test for this?

Reply:
> but it should be fine to add an entirely new (optional) field that happens to be an enum. 

This test won't fail in this scenario because adding a new field does not lead to a diff path ending with `enum`. The item addition will be detected at the parent level, not on the specific field property definition. I'll add a test for this for safety.

Reply:
Added the test 592e3c9d0eb9fa7bc875c7a8c4344675180a4983